### PR TITLE
[AddressLowering] Handle indexing into variadic tuples.

### PR DESCRIPTION
**Pull request by nate-chandler**
Although variadic values have their own conventions (`@pack_owned`, `@pack_guaranteed`) which have not yet been made direct in opaque values mode, variadic tuples use standard indirect conventions (`@in`, `@in_guaranteed`).  Consequently, in opaque values mode, a function with a variadic tuple parameter (e.g. `@in_guaranteed (repeat each T)`) will have a direct variadic tuple value in the entry block (`%tuple : @guaranteed $(repeat each T)`).

Elements are accessed from indirect variadic tuples via the `tuple_pack_element_addr` instruction.  For direct variadic tuples, the new instruction `tuple_pack_extract` is added here.  The new instruction is transformed by AddressLowering to `tuple_pack_element_addr`.  This is analogous to the situation with `tuple_extract` being transformed to `tuple_element_addr`.

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test macos platform

**Review by atrick — APPROVED** [review]
LGTM

**Comment by nate-chandler** [comment]
@swift-ci please test and merge

File: lib/SILOptimizer/Mandatory/AddressLowering.cpp [hunk 1]
```diff
@@ -3528,6 +3546,9 @@ class UseRewriter : SILInstructionVisitor<UseRewriter> {
   // Extract from an opaque tuple.
   void visitTupleExtractInst(TupleExtractInst *extractInst);
 
+  // Extract form an opaque pack tuple.
```
[the remark this answers is not in the document]
Comment (nate-chandler) [reply lib/SILOptimizer/Mandatory/AddressLowering.cpp]: `^M`

**Comment by nate-chandler** [comment]
@swift-ci please test linux platform

**Pull request merged**